src/lib.rs

```diff
@@ -341,6 +351,20 @@ impl<K: Eq + Hash + Sync + Clone, V: Sync + Clone> CHMSegment<K, V> {
         guard.unlinked(table);
     }
 
+    fn entries(&self) -> Vec<(K, V)> {
+        let mut xs = vec![];
+        let guard = epoch::pin();
+        let table = self.table.load(Relaxed, &guard).unwrap();
```

Comment:
Not sure if you're still interested in this PR given that I've abandoned you for so long - but I'm a bit concerned about these loads.  I think they will need to be Acquire to be safe, unless you have an explanation as to why not?